File: files/CH-160.py
import pandas as pd
import numpy as np

def place_knights_pattern(start_row=1, start_col=1):
    columns = np.arange(8)
    rows = ((start_row - 1 + columns - (start_col - 1)) % 8)
    board = np.full((8, 8), "", dtype=object)
    board[rows, columns] = "K"
    return pd.DataFrame(board)

print(place_knights_pattern(0, 5))

#    0  1  2  3  4  5  6  7
# 0                 K      
# 1                    K   
# 2                       K
# 3  K
# 4     K
# 5        K
# 6           K
# 7              K

File: files/100-199/160/CH-160.R
place_knights = function(start_row = 1, start_col = 1) {
  startY = start_row - 1
  startX = start_col - 1
  columns = 0:7
  rows = ((startY + columns - startX) %% 8) + 1
  board = matrix("", nrow = 8, ncol = 8)
  board[cbind(rows, columns + 1)] = "K"
  as.data.frame(board, stringsAsFactors = FALSE)
}

place_knights_pattern(2,6)

#   V1 V2 V3 V4 V5 V6 V7 V8
# 1              K         
# 2                 K      
# 3                    K   
# 4                       K
# 5  K                     
# 6     K                  
# 7        K               
# 8           K            